File: test-server.py
#!/usr/bin/env python3
"""
Simple test server to debug Chrome extension connectivity
"""

from flask import Flask, jsonify, request
from flask_cors import CORS

app = Flask(__name__)
CORS(app)  # Enable CORS for all routes

@app.route('/health', methods=['GET'])
def health():
    print("Health check requested")
    return jsonify({"status": "healthy", "test": "server working"})

@app.route('/processGeolocation', methods=['POST'])
def test_process():
    print("Process geolocation requested")
    print(f"Request data: {request.json}")
    
    return jsonify({
        "result": {
            "coordinates": {"lat": 40.7128, "lng": -74.0060},
            "location": "Test Location - New York City",
            "confidence": 0.95,
            "processing_time": 0.1,
            "model": "Test Model",
            "cost": 0,
            "tokensUsed": 0,
            "rawResponse": "Test response"
        }
    })

if __name__ == '__main__':
    print("🧪 Starting test server on port 8080...")
    print("This will help debug Chrome extension connectivity")
    app.run(host='0.0.0.0', port=8080, debug=True)

File: test-server.js
const express = require('express');
const cors = require('cors');
const app = express();
const port = 8081;

// Enable CORS for all routes
app.use(cors());
app.use(express.json({limit: '50mb'})); // Support large base64 images

app.get('/health', (req, res) => {
  console.log('Health check requested');
  res.json({
    status: 'healthy',
    test: 'Node.js server working',
    timestamp: new Date().toISOString()
  });
});

app.post('/processGeolocation', (req, res) => {
  console.log('Process geolocation requested');
  console.log('Request body keys:', Object.keys(req.body));
  
  res.json({
    result: {
      coordinates: {lat: 40.7128, lng: -74.0060},
      location: "Test Location - New York City",
      confidence: 0.95,
      processing_time: 0.1,
      model: "Test Node.js Server",
      cost: 0,
      tokensUsed: 0,
      rawResponse: "Test response from Node.js"
    }
  });
});

app.listen(port, '0.0.0.0', () => {
  console.log(`🧪 Test server running on http://localhost:${port}`);
  console.log('This will help debug Chrome extension connectivity');
});
